libs/ledger/src/chain/block_coordinator.cpp

```diff
@@ -1074,16 +1088,67 @@ BlockCoordinator::State BlockCoordinator::OnReset()
 {
   reset_state_count_->increment();
 
+  FETCH_LOG_INFO(LOGGING_NAME, "Reset condition");
+
   // trigger stake updates at the end of the block lifecycle
-  if (stake_)
+  if (stake_ && beacon_)
   {
+    Block const *block = nullptr;
+
     if (next_block_)
     {
-      stake_->UpdateCurrentBlock(*next_block_);
+      block = next_block_.get();
     }
     else if (current_block_)
     {
-      stake_->UpdateCurrentBlock(*current_block_);
+      block = current_block_.get();
+    }
+    else
+    {
+      FETCH_LOG_ERROR(LOGGING_NAME,
+                      "Unable to find a previously executed block when updating staking!");
+    }
+
+    // If we are on the head of the main chain, and it is a 'committee triggering' block
+    if (block && (*chain_.GetHeaviestBlock() == *block))
+    {
+      auto const block_number = block->body.block_number;
+
+      if ((block_number % AEON_PERIOD) == 0)
+      {
+        std::unordered_set<fetch::crypto::Identity> cabinet_member_list;
+        auto                                        snapshot = (*stake_).GetCurrentStakeSnapshot();
+        auto const current_stakers = snapshot->BuildCommittee(0, MAX_COMMITTEE_SIZE);
+
+        // TODO(HUT): switch to set for stake manager
+        for (auto const &staker : *current_stakers)
+        {
+          FETCH_LOG_DEBUG(LOGGING_NAME, "Adding staker: ", staker.identifier().ToBase64());
+          cabinet_member_list.insert(staker);
+        }
+
+        uint32_t threshold = uint32_t((cabinet_member_list.size() / 2) + 1);
+
+        FETCH_LOG_INFO(LOGGING_NAME, "Block: ", block_number,
+                       " creating new aeon. Periodicity: ", AEON_PERIOD, " threshold: ", threshold,
+                       " cabinet size: ", cabinet_member_list.size());
+
+        // Disallow multiple invocations
+        static bool did_genesis_already = false;
+
+        if (block_number != 0 || !did_genesis_already)
+        {
+          beacon_->StartNewCabinet(cabinet_member_list, threshold, block_number,
```

Comment:
`beacon_` should be of a interface class type so that we can mock it out

Reply:
Noted, will implement when doing unit tests for the block coord